### ISSUE TITLE
Do we have a plan to support macOS?

### ISSUE THREAD

**Issue by since2014** [microsoft/ntttcp-for-linux#32]
Do we have a plan to support macOS?

**Comment by simonxiaoss** [comment]
Bluecoca, has a PR to support macOS here: https://github.com/microsoft/ntttcp-for-linux/pull/19
I don't have a test environment to test the PR so I did not merge it yet.

**Comment by thanasisk** [comment]
@simonxiaoss If i test on FreeBSD, will you merge the request?

**Comment by simonxiaoss** [comment]
Yes, sure. I just want to have a graceful way to support multiple platforms. At least, the #ifdef in main line of the code is not acceptable to me. Maybe we should define abstract functions and then have different implementations for different platforms. 

I want the main line of the code is still readable.
